src/screens/Address/NewAddressScreen.tsx

```diff
@@ -80,14 +87,21 @@ const NewAddressScreen = ({ navigation }: ScreenProps) => {
   }
 
   return (
-    <>
-      <AddressForm
-        initialValues={initialValues}
-        onSubmit={handleGeneratePress}
-        onGroupPress={() => navigation.navigate('GroupSelectScreen')}
-      />
+    <Screen {...props}>
+      <NewAddressContextProvider>
```

Comment:
Do we really need to use a context here? 🤔 

Couldn't the group be passed directly from the GroupSelectModal to the parent?

Reply:
I am removing it in my follow up work on WC, I hope you don't mind if I don't touch this one.

Reply:
Also, I've just seen during my testing that the group doesn't seem to update properly.